### PR TITLE
Scaffold out the release process for this PR

### PR DESCRIPTION
**Pull request by justindbaur**
## 📔 Objective

Scaffold of my desired release process for this repo:

- Manually start the release process (`start-release.yml`):
  - Creates a branch for the release based on the `[major].[minor]` of the code currently checked in
  - Branch is named: `release/[PackageName]/[major].[minor]`
  - Push branch to remote, should automatically kick off `prerelease.yml`
  - Do a GA version bump (`0.1.0` -> `0.2.0`) and create PR to target
- Automatic prereleases (`prerelease.yml`)
  - Every commit to `release/*`
  - Runs `pack-and-release.yml` with input `prerelease: true`
  - Do a prerelease bump (`0.1.0-beta.1` -> `0.1.0-beta.2`) and create a PR to rc
- Do GA Release (`release.yml`)
  - Manually called when you consider your release candidate ready for GA
  - Does a hotfix version bump (`0.1.0` -> `0.1.1`) and create a PR to rc

`pack-and-release.yml` does the actual packing of the nuget package, publishes it to nuget, creates a GitHub release, and creates a version bump PR based on the given input. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/pack-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
+name: Publish
```
Comment (withinfocus) [inline .github/workflows/pack-and-release.yml:1]:
```suggestion
name: Pack and release
```

File: .github/workflows/pack-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      prerelease:
+        type: boolean
+
+jobs:
+  parse:
+    runs-on: ubuntu-22.04
+    outputs:
+      package: ${{ steps.parse-package.outputs.result }}
+    steps:
+      - name: Parse Package
```
Comment (withinfocus) [inline .github/workflows/pack-and-release.yml:15]:
```suggestion
      - name: Parse package
```

File: .github/workflows/pack-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                url: endpointString,
+                data: fs.readFileSync(file),
+              });
+
+              console.log(`Upload response: ${uploadResponse.status}`);
+            }
+
+            console.log("Finished creating release.");
+
+      - name: Do version bump
```
Comment (withinfocus) [inline .github/workflows/pack-and-release.yml:176]:
```suggestion
      - name: Bump version
```

File: .github/workflows/prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Do PreRelease
```
Comment (withinfocus) [inline .github/workflows/prerelease.yml:1]:
```suggestion
name: Perform prerelease
```

File: .github/workflows/prerelease.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+on:
+  push:
+    paths:
+      - "release/*"
+
+jobs:
+  do-prerelease:
```
Comment (withinfocus) [inline .github/workflows/prerelease.yml:9]:
```suggestion
  prerelease:
```

File: .github/workflows/prerelease.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    name: Do prerelease
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Pack & Release
```
Comment (withinfocus) [inline .github/workflows/prerelease.yml:13]:
```suggestion
      - name: Pack and release
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -1,200 +1,14 @@
-name: Release
-run-name: ${{ inputs.type }} release for ${{ inputs.package }}
+name: Do Release
```
Comment (withinfocus) [inline .github/workflows/release.yml:1]: ⛏️ Apply all my other OCD here too.

File: .github/workflows/start-release.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+permissions:
+  pull-requests: write
+  contents: write
+
+env:
+  PACKAGE_DIRECTORY: './extensions/${{ inputs.package }}'
+
+jobs:
+  start-release:
+    name: Create Release Candidate
```
Comment (withinfocus) [inline .github/workflows/start-release.yml:23]:
```suggestion
    name: Create release candidate
```

File: .github/workflows/start-release.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up .NET
+        uses: actions/setup-dotnet@3e891b0cb619bf60e2c25674b222b8940e2c1c25 # v4.1.0
+
+      # get current version
```
Comment (withinfocus) [inline .github/workflows/start-release.yml:32]: ⛏️ Remove as it's implied.

File: .github/workflows/start-release.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+              return;
+            }
+
+            const releaseCandidateBranch = `release/${packageName}/${versionParts[0]}.${versionParts[1]}`;
+
+            await exec.exec(`git checkout -b ${releaseCandidateBranch}`);
+
+            await exec.exec(`git push --set-upstream origin HEAD:${releaseCandidateBranch}`);
+
+      - name: Do version bump
```
Comment (withinfocus) [inline .github/workflows/start-release.yml:63]:
```suggestion
      - name: Bump version
```

File: .github/workflows/version-bump.yml [hunk 11]
```diff
@@ -0,0 +1,101 @@
+name: Version Bump
```
Comment (withinfocus) [inline .github/workflows/version-bump.yml:1]:
```suggestion
name: Bump version
```

File: .github/workflows/version-bump.yml [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      type:
+        description: The type of this release
```
Comment (withinfocus) [inline .github/workflows/version-bump.yml:7]:
```suggestion
        description: Type of release
```

**Review by withinfocus — COMMENTED** [review]
Didn't read the script blocks too thoroughly since I believe this is just moving things around. Bunch of opinionated comments on nouns and verbs for all the steps otherwise.

File: .github/workflows/release.yml (hunk 7 above)
Comment (justindbaur) [reply .github/workflows/release.yml]: Is d0ec729dcbb32074973924ee893535d65b99906f good?

File: .github/workflows/start-release.yml [hunk 13]
```diff
@@ -0,0 +1,66 @@
+name: Release
+run-name: Starting release for ${{ inputs.package }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+        description: Which package to release
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply .github/workflows/start-release.yml:8]: I'll add this to the next fix PR that I will probably need.

**Pull request merged**